### ISSUE TITLE
[feature request] proper DRF support (MoneyField)

### ISSUE THREAD

**Issue by C0DK** [django-money/django-money#550]
Currently it is only possible to use django money with DRF if the money value is actually from a concrete model - meaning it is impossible to return a money field from a property, for instance - I have a lot of dynamic calculations of money-fields which i want to return with REST, however this is rather difficult.

One could (presumably) relatively easy create a custom DRF field (and I might look into this) for this exact purpose, that returns a JSON object, however this would naturally conflict with current usage of django money. I'm not sure if you could make a DRF field that creates two fields. My current best case is to return decimal and have a secondary property return the currency. This could probably be streamlined, but i would suggest adding a MoneyField as an optional serializer fieldtype, that can be utilized by explicitly utilizing it, and then potentially slowly making the current solution obsolete/deprecated, howveer I don't know if there is an actual reason for it to returning two fields (other than it being the DB representation of the money fields). 

This is just my 5 cent, and you should naturally ignore it if you hate it - I might, when I have some time, implement something and create a PR if this is a feature that would be interesting to have.

**Comment by benjaoming** [comment]
Hi @C0DK  - thanks for opening an issue to improve DRF support

I'm not sure what you mean by "proper", is there a more DRF-friendly implementation of a custom database field that you could point to? A PR is absolutely welcome.

We've covered DRF support through serialization if that helps? https://github.com/django-money/django-money#usage-with-django-rest-framework

**Comment by C0DK** [comment]
Yes, but to the best of my knowledge i cannot serialize af property for instance. I don't seem to be able to find the actually DRF field class that i can use neither, so unless my Money object is stored as an actual field on the model, then I am not able to use it.

**Comment by stianjensen** [comment]
You should be able to import this field and use it explicitly on your serializer:
```
from djmoney.contrib.django_rest_framework import MoneyField

class Serializers(serializers.Serializer):
    my_computed_prop = MoneyField(max_digits=10, decimal_places=2)
```

**Comment by stianjensen** [comment]
I read over your question once more now. Still not _quite_ sure what you want to happen, but I _think_ you want to output the value and the currency of your custom class property that returns a money object as two separate fields on your serializer (like what happens by default for a model field).

The only way I've been able to do that is something like this:
```
class MyClass:
    @property
    def my_computed_prop(self):
        return Money(10, 'EUR')


from djmoney.contrib.django_rest_framework import MoneyField

class Serializers(serializers.Serializer):
    my_computed_prop = MoneyField(max_digits=10, decimal_places=2)
    my_computed_prop_currency = CharField(source='my_computed_prop.currency')
```

I'm not sure if you can implement a custom field in drf that returns data that will be exposed as two fields on the serializer (at least not without also altering the Serializer class itself). You could implement an alternative to MoneyField that generates a nested representation like this, though:
```
{"my_computed_prop": {"amount": "10.00", "currency": "EUR"}}
```

**Comment by matthewgoslett** [comment]
Thanks @stianjensen - works for me.

For what it's worth, I have the same use case.

I have a `Cart` model which has the following property:

```python
@property
def net_amount(self):
    return self.gross_amount - self.discount
```

I want to expose this calculated property in my serializer.

The following works for me:
```python
net_amount = MoneyField(max_digits=10, decimal_places=2)
net_amount_currency = serializers.CharField(source="net_amount.currency")
```

**Comment by C0DK** [comment]
@stianjensen sorry, yes that was exactly my concern. My last message was lacking - sorry. But your example:


> The only way I've been able to do that is something like this:
> 
> ```
> class MyClass:
>     @property
>     def my_computed_prop(self):
>         return Money(10, 'EUR')
> 
> 
> from djmoney.contrib.django_rest_framework import MoneyField
> 
> class Serializers(serializers.Serializer):
>     my_computed_prop = MoneyField(max_digits=10, decimal_places=2)
>     my_computed_prop_currency = CharField(source='my_computed_prop.currency')
> ```

Is exactly the scenario i have. It would be great if the default simply was to return a dict rather than have two fields, but that is probably not going to be changed within your project. But that solution sort of work, thank you.

You are welcome to close the issue, but maybe the example should be documented in the readme.

**Comment by sdarmofal** [comment]
It will be very nice to have documented in the readme, that it's necessary to add the money field with the **_currency** suffix into serializer fields. It took me a while to figure out how to serve the currency through the API.

**Comment by benjaoming** [comment]
@sdarmofal :green_circle:  :green_circle:  :green_circle:  :green_circle:  :green_circle:  :green_circle: 

:D

More and better documentation is always welcome!!

**Comment by gabn88** [comment]
It would be nice to have the OPTIONS request list all the possible currencies and make it a ChoiceField instead of a CharField. That way it would be treated as an Enum by OpenAPI generators and it would be much easier to use for frontent engineers.

Posting it here, since that in my opinion has to do with 'proper' DRF support. I do agree though that a nested representation: 

```
   price = MoneyField()
```
that would return

```
   price = {'amount': '10.00', 'currency': 'EUR'}
```
Is preferrable over the current implementation of adding an extra _currency field. On the other hand the current implementation is closer to the database representation, which has the advantage of simplicity (and simple > complex).
